feat: implement coin-based payment system for premium downloads

Matinj031 opened the pull request:
## 🎯 Overview

This PR implements a complete coin-based payment system for premium downloads, specifically targeting 2025 past papers that require 5 coins to access. The feature includes visual feedback, API integration, and user-friendly payment flows.

## ✨ Features Added

### 🪙 Coin Payment Modal

- Interactive payment confirmation dialog
- Real-time balance display and validation
- Insufficient balance error handling
- Clean UI with coin-themed styling

### 🎬 Coin Consumption Animation

- Visual feedback during coin deduction using Lottie animations
- "×5" multiplier indicator to clearly show coins being consumed
- Smooth fade-in animations for better UX
- Full-screen overlay with semi-transparent background

### 🔌 Backend API Integration

- Integrated `/api/v1/game/coins/consume` endpoint
- Real coin deduction instead of simulation
- Automatic balance refresh after successful transactions
- Proper error handling and user feedback

Matinj031 committed:
feat: add coin consumption animation component with 5 multiplier indicator

- Add CoinConsumptionAnimation.vue component for visual feedback during coin deduction
- Display large coin animation using Lottie with lose.json animation
- Include 5 multiplier indicator to show number of coins being consumed
- Add fade-in scale animation for smooth multiplier appearance
- Position multiplier with golden styling and text shadow for visibility

Matinj031 committed:
feat: add coin payment modal for premium downloads

- Add CoinPaymentModal.vue component for handling 5-point premium downloads
- Display user balance and payment confirmation interface
- Include insufficient balance validation and error messaging
- Add confirm/cancel actions with loading states
- Style with orange coin theme and responsive layout

Matinj031 committed:
feat: integrate coin consumption API with backend

- Add consumeCoins function to call /api/v1/game/coins/consume endpoint
- Update deductCoins to use real API instead of simulation
- Send points parameter in request body as required by backend
- Auto-refresh balance after successful coin consumption
- Maintain backward compatibility with existing deductCoins interface

Matinj031 committed:
feat: add coin lose animation asset

- Add lose.json Lottie animation file for coin consumption visual feedback
- Animation shows coins being consumed/lost during payment process

Matinj031 committed:
style: fix formatting in ContentInfo component

- Add proper spacing after closing brace in script section
- Maintain consistent code formatting standards

Matinj031 committed:
feat: add coin payment support to MobileOrder component

- Add requiresCoinPayment prop to handle premium download logic
- Update download button to show coin icon for premium content
- Integrate coin payment flow for 2025 past papers requiring payment

Matinj031 committed:
minor changes: remove logs, adjustment animation timing

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into feat/loseCoin

Matinj031 committed:
chore: add .vscode/ to .gitignore

- Prevent VS Code settings from being tracked in repository
- Keeps local development configurations private

Matinj031 committed:
feat: implement coin consumption animation component

- Add overlay animation for coin deduction feedback
- Include Lottie animation with lose.json asset
- Display coin count with fade-in-scale animation
- Emit animation-complete event for parent handling
- Add fallback timeout for animation completion

Matinj031 committed:
feat: add coin payment modal for premium downloads

- Create modal for 5-coin premium paper downloads
- Display user balance and insufficient funds warning
- Add confirm/cancel actions with loading states
- Include wallet icon and premium download messaging
- Integrate with formatNumber composable for balance display

Matinj031 committed:
feat: integrate coin payment system for 2025 papers

- Add coin balance checking and payment flow for 2025 files
- Implement coin deduction with animation feedback
- Show coin cost indicators on download buttons
- Handle authentication requirements for premium content
- Add loading states and error handling for coin transactions
- Integrate coin consumption animation after successful payment

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into feat/loseCoin

Matinj031 committed:
feat: remove price

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into feat/loseCoin

Matinj031 committed:
feat: implement selective coin payment for 2025 files

- Add requiresCoinPaymentForFile() function to determine which file types need coins
- 2025 question papers (PDF/Word) remain free
- 2025 mark schemes and extra files require 5 coins
- Update download logic to use selective coin payment
- Fix FileSaver.js implementation for proper download dialog
- Add success message for coin-based downloads
- Remove coin indicators from question paper buttons for 2025 files

Matinj031 committed:
fix: add default function for requiresCoinPaymentForFile prop

- Add proper default function to requiresCoinPaymentForFile prop
- Ensures component works correctly when function is not provided
- Maintains backward compatibility

Matinj031 committed:
feat: add authentication guards to useCoinBalance composable

- Add authentication checks in fetchBalance(), consumeCoins(), and deductCoins()
- Prevent unauthorized API calls to coin-related endpoints
- Return early with error message when user is not authenticated
- Remove unused interfaces and variables for cleaner code

Matinj031 committed:
feat: integrate login dialog for premium file downloads

- Replace toast messages with login dialog when user is not authenticated
- Use existing openAuthDialog() pattern for consistent UX across app
- Add dynamic button text showing 'Login to Download' for premium files
- Add authentication watcher to refresh coin balance after login
- Improve error handling for 401 responses to trigger login dialog
- Add disabled state for download buttons when authentication required
- Enhance security by preventing unauthorized access to premium content

sanaderi commented on composables/useCoinBalance.ts:
```diff
@@ -0,0 +1,144 @@
+   */
+  const fetchBalance = async (): Promise<number> => {
+    // Check authentication first
+    if (!auth.isAuthenticated.value) {
```
Remove this condition; there’s no need to handle auth manually.

sanaderi commented on composables/useCoinBalance.ts:
```diff
@@ -0,0 +1,144 @@
+   */
+  const consumeCoins = async (points: number): Promise<boolean> => {
+    // Check authentication first
+    if (!auth.isAuthenticated.value) {
```
Remove this condition; there’s no need to handle auth manually.

sanaderi commented on components/paper/detail/ContentInfo.vue:
```diff
@@ -302,51 +372,281 @@ const isFree = computed(() => {
+
+// Fetch coin balance on component mount
+onMounted(async () => {
+  if (auth.isAuthenticated.value && requiresCoinPayment.value) {
```
We don’t need to check the balance on mounted—it causes an extra request. We only need the user’s balance when it’s actually required.

sanaderi commented on components/paper/detail/ContentInfo.vue:
```diff
@@ -302,51 +372,281 @@ const isFree = computed(() => {
+})
+
+// Watch for authentication changes to refresh coin balance
+watch(() => auth.isAuthenticated.value, async (isAuthenticated) => {
```
We don’t need to watch balance—it causes an extra request. We only need the user’s balance when it’s actually required.

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into feat/loseCoin

Matinj031 committed:
feat(coinPaymentModal): add $GET icon for the confirm modal, improve some css, styles

Matinj031 committed:
refactor(ContentInfo): remove fetching balance from onMounted   hook, remove unnecessary watchers

Matinj031 committed:
fix(useCoinBalance): remove extras auth checks, clean up code

Matinj031 committed:
fix: enable all download buttons, show login on click for auth

- Remove button disabled states for unauthenticated users
- Consistent auth flow: show login form on any button click
- Remove redundant auth checks causing extra API requests

sanaderi committed:
fix(paper): update primium file alert sentense
